src/sentry/workflow_engine/migration_helpers/issue_alert_dual_write.py

```diff
@@ -202,10 +202,26 @@ def delete_migrated_issue_alert(rule: Rule) -> int | None:
     workflow: Workflow = alert_rule_workflow.workflow
     workflow_id = workflow.id
 
-    delete_workflow(workflow)
-    alert_rule_workflow.delete()
+    try:
+        delete_workflow(workflow)
+    except DataConditionGroup.DoesNotExist:
+        logger.exception(
+            "Deleting workflow but DCG does not exist",
+            extra={"workflow_id": workflow_id, "rule_id": rule.id},
+        )
 
-    return workflow_id
+    try:
+        workflow.refresh_from_db()
+    except Workflow.DoesNotExist:
```

Comment:
Should you have a test for this case?

Reply:
let me test for the return value specifically, it should be tested in `test_delete_issue_alert__handle_race_condition_deleted`